File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        dp=[0]*len(cost)
        if len(cost)==0:
            return 0
        if len(cost)==1:
            return cost[0]
        dp[0]=cost[0]
        dp[1]=cost[1]
        for i in range(2,len(cost)):
            dp[i]=cost[i]+min(dp[i-1],dp[i-2])
        return min(dp[len(cost)-1],dp[len(cost)-2])
        
        

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        /*
        ArrayList<Integer> dp =new ArrayList<>();
        dp.add(0);
        dp.add(0);

        for(int i=2;i<cost.length+1;i++){
            dp.add(Math.min(dp.get(i-1)+cost[i-1],dp.get(i-2)+cost[i-2]));
        }
        return dp.get(dp.size()-1);*/
        int prev=0;
        int current=0;
        int temp=0;
        for(int i=2;i<cost.length+1;i++){
            temp=prev;
            prev=current;
            current=Math.min(temp+cost[i-2],current+cost[i-1]);
        }
        return current;
    }
}